core/src/commitment.rs

```diff
@@ -123,22 +145,26 @@ impl BlockCommitmentCache {
     }
 
     pub fn is_confirmed_rooted(&self, slot: Slot) -> bool {
-        self.get_block_commitment(slot)
-            .map(|block_commitment| {
-                (block_commitment.get_rooted_stake() as f64 / self.total_stake as f64)
-                    > VOTE_THRESHOLD_SIZE
-            })
-            .unwrap_or(false)
+        slot <= self.largest_confirmed_root()
+            && (self.blockstore.is_root(slot)
+                || (self.bank.status_cache_ancestors().contains(&slot)
+                    && self
```

Comment:
I think this last check is not necessary because if the the `largest_confirmed_root()` must be an ancestor of `self.bank` right, since it was built from state using `self.bank`